modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java

```diff
@@ -178,6 +194,33 @@ public void validateUpdate(SqlUpdate call) {
         syncSelectList(select, call);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    protected void checkTypeAssignment(
+            SqlValidatorScope sourceScope,
+            SqlValidatorTable table,
+            RelDataType sourceRowType,
+            RelDataType targetRowType,
+            SqlNode query
+    ) {
+        boolean coerced = false;
+
+        if (config().typeCoercionEnabled()) {
+            if (SqlTypeUtil.equalAsStructSansNullability(typeFactory,
```

Comment:
Special casing for BIGINT does not fix the problem with BIGINT literals in apache calcite (there is a bug in calcite). Add this code should be removed.

Reply:
Well, I am mistaken about a bug in calcite. I think but special casing of BIGINT does not belong to this PR I removed the code related to special casing of BIGINT observed no changes in code behaviour.

Reply:
removing of this part leads to testInsertValueOverflow failure, something wrong with Idea caching or rebuilding, first run of such a test with commented lines gives me correct tests passing but second is already not, the same for IgniteSqlToRelConvertor part.